baiji/util/tempfile.py

```diff
@@ -7,10 +7,13 @@
 '''
 from __future__ import absolute_import
 import os
+from baiji.util.shutillib import mkdir_p
 
 def _override_tmp(fn, args, kwargs):
     if 'dir' not in kwargs:
         kwargs['dir'] = os.environ.get('BAIJI_TMP', None)
+        if kwargs['dir'] is not None:
+            mkdir_p(kwargs['dir'])
```

Comment:
No. `mkstmp` is supposed to honor the `TMP` env var and let you put your temp files somewhere else based on that. But that doesn't actually work on all platforms, which is why `_override_tmp` exists. The problem we ran into is that if the temp location is overridden using the `dir` argument, then it assumes that you know what you're doing and that it exists. That's the problem we're trying to fix here. Everything definitely works without this var set, since that's the behavior we've been using all along.